stdlib/src/builtin/string.mojo

```diff
@@ -609,6 +609,64 @@ struct String(Sized, Stringable, IntableRaising, KeyElement, Boolable):
         """
         return not (self == other)
 
+    @always_inline
+    fn __gt__(self, other: String) -> Bool:
+        """Compares two Strings if self greater than the other.
+
+        Args:
+            other: The rhs of the operation.
+
+        Returns:
+            True if self are greater than other string.
+        """
+        if self == other:
```

Comment:
At the worst case, this line is already one pass on every bytes in the string. It should be possible to get the result of `__gt__` with just a size comparison and a `memcmp`, which does, at the worst, one read on every byte